### DISCUSSION TITLE
data not load always from the first time

### DISCUSSION THREAD

**Discussion by itsalimanuel** [nuxt/nuxt#10765] [Questions (Nuxt 2)]
Hello guys, I made a Website using **Nuxt**,all perfect except data load ( from squidex api), the data not loaded always and sometimes i need to refresh the website many times to get the data.
I tried :
```javascript
methods: {
    getData() {
      axios.get('https://cloud.squidex.io/api/content/username/homepage').then(res => {
        if (res.data) {
          let data = res.data.items[0].data
          this.home = {
            name: data.name,
          }
        } else {
          this.home = []
        }
      })
    },

  },
  created() {
    this.getData()
  }

and i tried :

  methods: {
   async getData() {
     await  axios.get('https://cloud.squidex.io/api/content/username/homepage').then(res => {
        if (res.data) {
          let data = res.data.items[0].data
          this.home = {
            name: data.name,
          }
        } else {
          this.home = []
        }
      })
    },

  },
  created() {
    this.getData()
  }
  
but the same result data not loaded always!

**Comment by ghost** [comment]
The created-lifecycle is called for both server and client. And component state (data-option) is seperate for each instance. Meaning if you fetch data inside the created-lifecycle, that data will first be "lost" and then fetched again:
https://nuxtjs.org/docs/concepts/nuxt-lifecycle/

I'm assuming you're using Nuxt2 from your use of the Options-Api.
Nuxt has its own lifecycle methods for enabling severside- fetching and keeping state. asyncData() (return value sets/merges with data-option) for page-components and fetch() for any other component (executes on server on page-request and on client-side navigation, if you don't use    don't use set the render target to "static", and caches the components state for client-side). 

And then theres vue's serverPrefetch: similar to fetch() but will only be executed on the server, meaning will only work in Nuxt if your render-target is "server" and a server is running your Nuxt-App.
I wouldn't recommend to do data-fetching in any other lifecycle.
